src/replica/Registry.cc

```diff
@@ -46,6 +50,47 @@ LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.Registry");
 
 string context(string const& func) { return "REGISTRY " + func + " "; }
 
+/**
+ * @brief Get the FQDN of the current host.
+ * @note If there will be more than one canonical name associated with the host
+ *   then  the result will containe them all separated by comma, such as in:
+ * @code
+ *   <fqdn-1>,<fqdn-2>,...
+ * @endcode
+ * @return The FQDN (or FQDNs)
+ * @throws std::runtime_error In case if the information couldn't be retreived.
+ */
+string get_current_host_fqdn() {
+    // Get the short name of the current host first.
+    boost::system::error_code ec;
+    string const hostname = boost::asio::ip::host_name(ec);
+    if (ec.value() != 0) {
+        throw runtime_error("Registry::" + string(__func__) +
+                            " boost::asio::ip::host_name failed: " + ec.category().name() + string(":") +
+                            to_string(ec.value()) + "[" + ec.message() + "]");
+    }
+
+    // Get the host's FQDN(s)
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_UNSPEC;     /* either IPV4 or IPV6 */
+    hints.ai_socktype = SOCK_STREAM; /* IP */
+    hints.ai_flags = AI_CANONNAME;   /* canonical name */
+    struct addrinfo* info;
+    int const retCode = getaddrinfo(hostname.data(), "http", &hints, &info);
+    if (retCode != 0) {
+        throw runtime_error("Registry::" + string(__func__) +
+                            " getaddrinfo failed: " + gai_strerror(retCode));
+    }
```

Comment:
There could be temporary failures for which `getaddrinfo` returns `EAI_AGAIN` code and you are supposed to re-try again.